utils/sample_qc.py

```diff
@@ -363,99 +363,114 @@ def infer_sex(
+    x_ploidy_cutoffs: [upper cutoff for single X, (lower cutoff for double X, upper cutoff for double X), lower cutoff for triple X]
+    y_ploidy_cutoffs: [lower cutoff for single Y, upper cutoff for single Y, lower cutoff for double Y]
+
+    Uses the normal_ploidy_cutoff parameter to determine the ploidy cutoffs for XX and XY karyotypes. 
+    Uses the aneuploidy_cutoff parameter to determine the cutoffs for sex aneuploidies. 
+    
+    Note that f-stat is used only to split the samples into roughly 'XX' and 'XY' categories and is not used in the final karyotype annotation.
+
+    :param Table ht: Table with f_stat and sex chromosome ploidies
+    :param float f_stat_cutoff: f-stat to roughly divide 'XX' from 'XY' samples. Assumes XX samples are below cutoff and XY are above cutoff.
+    :param int normal_ploidy_cutoff: Number of standard deviations to use when determining sex chromosome ploidy cutoffs for XX, XY karyotypes.
+    :param int aneuploidy_cutoff: Number of standard deviations to use when sex chromosome ploidy cutoffs for aneuploidies.
+    :return: Tuple of ploidy cutoff lists: ([chrX cutoffs], [chrY cutoffs])
+    :rtype: tuple
+    """
+    # Group sex chromosome ploidy table by f_stat cutoff and get mean/stdev for chrX/Y ploidies
+    sex_stats = ht.aggregate(
+        hl.agg.group_by(
+            hl.cond(ht.f_stat < f_stat_cutoff, "xx", "xy"),
+            hl.struct(
+                    x=hl.agg.stats(ht.chrX_ploidy),
+                    y=hl.agg.stats(ht.chrY_ploidy)
+            )
+        )
+    )
+
+    cutoffs = (
+                [
+                    sex_stats["xy"].x.mean + (normal_ploidy_cutoff * sex_stats["xy"].x.stdev),
+                    (sex_stats["xx"].x.mean - (normal_ploidy_cutoff * sex_stats["xx"].x.stdev), sex_stats["xx"].x.mean + (normal_ploidy_cutoff * sex_stats["xx"].x.stdev)),
+                    sex_stats["xx"].x.mean + (aneuploidy_cutoff * sex_stats["xx"].x.stdev)
+                ],
+                [
+                    sex_stats["xx"].y.mean + (normal_ploidy_cutoff * sex_stats["xx"].y.stdev),
+                    sex_stats["xy"].y.mean + (normal_ploidy_cutoff * sex_stats["xy"].y.stdev), 
+                    sex_stats["xy"].y.mean + (aneuploidy_cutoff * sex_stats["xy"].y.stdev)
+                ]
+            )
+ 
+    logger.info(f"XX stats: {sex_stats['xx']}") 
+    logger.info(f"XY stats: {sex_stats['xy']}")
+    return cutoffs
+
+
 def get_sex_expr(
     chr_x_ploidy: hl.expr.NumericExpression,
     chr_y_ploidy: hl.expr.NumericExpression,
-    f_stat: hl.expr.NumericExpression,
-    xx_x_ploidy_cutoffs: Tuple[float, float] = (1.4, 2.25),
-    xy_x_ploidy_cutoffs: Tuple[float, float] = (0.5, 1.4),
-    xx_y_ploidy_cutoff: float = 0.1,
-    xy_y_ploidy_cutoffs: Tuple[float, float] = (0.15, 1.2),
-    yy_y_ploidy_cutoff: float = 1.3,
-    xxx_x_ploidy_cutoff:float = 2.5,
-    f_stat_female_cutoff: float = -0.2,
-    f_stat_male_cutoff: float = 0.2
+    x_ploidy_cutoffs: list,
```

Comment:
same as `tuple` up  there, these need to be typing classes (so `list`- > `List`). That said, if  you agree with my proposed changes, then it should be `Tuple[float, Tuple[float, float], float]` for `x_ploidy_cutoffs` and `Tuple[Tuple[float, float], float]` for `y_ploidy_cutoffs`